main.go

```diff
@@ -456,177 +467,170 @@ func makeSecretChanges(kubeClient *k8s.Client, secret *corev1.Secret, initiator
+	eventResp, exist, err := isEventExist(kubeClientset, secret.Namespace, eventName)
 
 	if exist == "error" {
 		return err
 	}
 
 	if exist == "found" {
-		count = *eventResp.Count + 1
-		eventResp.Type = &eventType
-		eventResp.Action = &action
-		eventResp.Reason = &reason
-		eventResp.Message = &message
-		eventResp.Count = &count
-		eventResp.LastTimestamp.Seconds = &secs
-		err = kubeClient.Update(context.Background(), &eventResp)
+		count = eventResp.Count + 1
+		eventResp.Type = eventType
+		eventResp.Action = action
+		eventResp.Reason = reason
+		eventResp.Message = message
+		eventResp.Count = count
+		eventResp.LastTimestamp = metav1.NewTime(now)
+		_, err = kubeClientset.CoreV1().Events(secret.Namespace).Update(eventResp)
 
 		if err != nil {
-			log.Error().Msgf("Event %v.%v - Updating Event has an error.\n\t%s", *eventResp.Metadata.Name, *eventResp.Metadata.Namespace, err.Error())
+			log.Error().Msgf("Event %v.%v - Updating Event has an error.\n\t%s", eventResp.Name, eventResp.Namespace, err.Error())
 			return err
 		}
 
-		log.Info().Msgf("Event %v.%v - has been updated successfully...", *eventResp.Metadata.Name, *eventResp.Metadata.Namespace)
+		log.Info().Msgf("Event %v.%v - has been updated successfully...", eventResp.Name, eventResp.Namespace)
 		return
 	}
 
-	event := &corev1.Event{
-		Metadata: &metav1.ObjectMeta{
-			Name:      &eventName,
-			Namespace: secret.Metadata.Namespace,
-			CreationTimestamp: &metav1.Time{
-				Seconds: &secs,
-			},
-			Labels: secret.Metadata.Labels,
+	event := &v1.Event{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:              eventName,
+			Namespace:         secret.Namespace,
+			CreationTimestamp: metav1.NewTime(now),
+			Labels:            secret.Labels,
 		},
-		FirstTimestamp: &metav1.Time{
-			Seconds: &secs,
+		FirstTimestamp: metav1.NewTime(now),
+		LastTimestamp:  metav1.NewTime(now),
+		Type:           eventType,
+		Action:         action,
+		Reason:         reason,
+		Message:        message,
+		Count:          count,
+		Source: v1.EventSource{
+			Component: eventSource,
 		},
-		LastTimestamp: &metav1.Time{
-			Seconds: &secs,
+		InvolvedObject: v1.ObjectReference{
+			Namespace: secret.Namespace,
+			Kind:      kind,
 		},
-		Type:    &eventType,
-		Action:  &action,
-		Reason:  &reason,
-		Message: &message,
-		Count:   &count,
-		Source: &corev1.EventSource{
-			Component: &eventSource,
-		},
-		InvolvedObject: &corev1.ObjectReference{
-			Namespace: secret.Metadata.Namespace,
-			Kind:      &kind,
-		},
-		EventTime: &metav1.MicroTime{
-			Seconds: &secs,
-		},
-		ReportingComponent: &reportingController,
-		ReportingInstance:  &reportingInstance,
+		EventTime:           metav1.NewMicroTime(now),
+		ReportingController: reportingController,
+		ReportingInstance:   reportingInstance,
 	}
 
-	err = kubeClient.Create(context.Background(), event)
-
+	_, err = kubeClientset.CoreV1().Events(secret.Namespace).Create(event)
```

Comment:
Now that the event object has the namespace set I think you can either use `.Events(event.Namespace)` or `.Events("")` to ensure it's slightly more robust and unlikely to go wrong if you every copy this elsewhere.